lib/winston-logstash-udp.js

```diff
@@ -46,7 +46,13 @@ util.inherits(LogstashUDP, winston.Transport);
 winston.transports.LogstashUDP = LogstashUDP;
 
 LogstashUDP.prototype.connect = function() {
+    var self = this;
     this.client = dgram.createSocket('udp4');
+
+    this.client.on('error', function(err) {
+        // Skipping DNS errors
+        self.emit('error', err);
```

Comment:
Exactly. I'm not trying to suppress the errors. In old version of node, it appears that even if you're using the dgram socket as a send-only connection, you still need to have an error handler defined on the socket listener. If there's not an error listener on the socket, you'll get an unhandled exception propagated to the root. With this error listener, we avoid the top level exception. The throwned exception in dns.js has been fix in recent node version only.